src/io/linux.zig

```diff
@@ -1367,6 +1367,76 @@ pub const IO = struct {
         self.enqueue(completion);
     }
 
+    pub const Event = posix.fd_t;
+    pub const INVALID_EVENT: Event = -1;
+
+    pub fn open_event(self: *IO) !Event {
+        _ = self;
+
+        // eventfd initialized with no (zero) previous write value.
+        const event_fd = posix.eventfd(0, linux.EFD.CLOEXEC) catch |err| switch (err) {
```

Comment:
Pre-existing, but we probably shouldn't go through `posix` here, for two reasons:

* even if libc is linked we'd rather do syscalls directly, to not mess up with errno
* we _know_ we are linux, we don't need cross-platform posix abstraction!

Reply:
>The thing is, we'd need to handle errno for linux.eventfd(...) ourselves.


I don't think so? The errno is the libc thing, not the syscall thing. sys calls just return an error code as a return value as any non-insane API would do, and then libc wrapper helpfully stuffs the error code into the `errno` thread local. 

So if we use `linux.xxx` API, we don't have to touch `errno`. 

Or are you talking specifically about converting an `u32` return number to an Error value? I _think_ `std.os.linux.E.init(code)` would do that?

Reply:
> Or are you talking specifically about converting an u32 return number to an Error value? I think std.os.linux.E.init(code) would do that?

Yes, I mean making our `io/linux.zig` to switch over `E` instead of handling the `error` returned by the posix abstraction (not only for `eventfd`, but for all other places we use a posix call such as `write`, `open`, `close`, `fsync`, etc).